drivers/arm_scmi/shmem.h

```diff
@@ -0,0 +1,37 @@
+/*
```

Comment:
I think it will be better to move arm_scmi to drivers/firmware because of the following points:
1) I think it's better to follow linux pattern.
2) It will be easier to add another protocols such as scpi
3) You already put includes inside include/zephyr/drivers/firmware folder

Reply:
done